npe2/manifest/utils.py

```diff
@@ -74,10 +74,12 @@ def get_callable(
             _registry = PluginManager.instance().commands
         return _registry.get(self.command)
 
-    @property
-    def plugin_name(self: ProvidesCommand):
-        # takes advantage of the fact that command always starts with manifest.name
-        return self.command.split(".")[0]
+    @cached_property
+    def plugin_name(self: ProvidesCommand) -> str:
+        from .._plugin_manager import PluginManager
```

Comment:
@nclack, I don't love what I'm having to do here... but this gets back to https://github.com/napari/npe2/issues/28

lemme know if you have any better ideas

Reply:
I don't love it either 😄 It does work 🥇 

Among other things, it's weird to me that asking for the plugin name depends on whether the command is registered with a plugin manager. 

Brainstorming options:
1. add some attribute to `Executable` that tracks the owning `PluginManager` or the `plugin_name`. It could get added during `register_command().` It would mean that this function doesn't work for a plugin that hasn't been registered. 
2. Parse command id's somehow during pydantic validation transforming `complex.plugin.name.command.id` to `(complex.plugin.name,command.id)`. The idea is to parse it while we still have the plugin name around. 
3. Change the command id spec so we use a distinct separator between plugin name and command id - `complex.plugin.name:command.id`.
4. Iterate through all the '.' splits and check for valid package names. This just seems actively bad but who knows.

Option 1 might be my favorite if it sounds workable.

Depending on how these ideas hit you, I'd approve the current approach with an issue calling out the use of the global PluginManager to fix later.